library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -446,17 +447,24 @@ public void pauseBatch(long id) {
     }
 
     public void resumeBatch(long id) {
-        ContentValues values = new ContentValues();
-        values.put(DownloadContract.Downloads.COLUMN_CONTROL, DownloadsControl.CONTROL_RUN);
-        values.put(DownloadContract.Downloads.COLUMN_STATUS, DownloadStatus.PENDING);
+        ContentValues values = getResumeContentValues();
         String where = COLUMN_BATCH_ID + "= ? AND " + DownloadContract.Downloads.COLUMN_STATUS + " != ?";
-        String[] selectionArgs = {String.valueOf(id), String.valueOf(DownloadStatus.PENDING)};
+        String[] selectionArgs = {String.valueOf(id), String.valueOf(DownloadStatus.SUCCESS)};
```

Comment:
Seems ok, my fix was to ensure that we only change the minimum number of values, added the STATUS != PENDING won't change the result but will avoid to change some unnecessary rows.

I'm ok with this